py/frequenz/client/weather/_types.py

```diff
@@ -263,19 +263,161 @@ def to_ndarray_vlf(
 
             # Check if the array shape matches the number of filtered times, locations
             # and features
-            if array.shape[0] != len(validity_times_indexes):
+            if validity_times is not None and array.shape[0] != len(validity_times):
                 print(
-                    f"Warning:  The count of validity times in the "
-                    f"array({array.shape[0]}) does not match the expected time "
-                    f"filter count ({validity_times_indexes}."
+                    (
+                        f"Warning:  The count of validity times in the "
+                        f"array({array.shape[0]}) does not match the expected time "
+                        f"filter count ({validity_times_indexes}."
+                    )
                 )
-            if array.shape[1] != len(location_indexes):
+            if locations is not None and array.shape[1] != len(locations):
                 print(
                     f"Warning:  The count of location in the "
                     f"array ({array.shape[1]}) does not match the expected location "
                     f"filter count ({location_indexes})."
                 )
-            if array.shape[2] != len(feature_indexes):
+            if features is not None and array.shape[2] != len(features):
+                print(
+                    f"Warning: The count of features ({array.shape[2]}) does not "
+                    f"match the feature filter count ({feature_indexes})."
+                )
+
+        except Exception as e:
+            raise RuntimeError("Error processing forecast data") from e
+
+        return array
+
+    # pylint: disable=too-many-locals,too-many-branches,too-many-statements
+    def to_ndarray_vlf_refined(
```

Comment:
It's not clear what refined means. Why do we even need both? We can just change the validity_times type in the original `to_ndarray_vlf`  as suggested in #77